flexmeasures/data/models/planning/storage.py

```diff
@@ -739,7 +712,19 @@ def deserialize_flex_config(self):
         if self.flex_model is None:
             self.flex_model = {}
 
-        self.flex_context = FlexContextSchema().load(self.flex_context)
+        # self.flex_context overrides self.asset.flex_context, which overrides self.asset.parent.flex_context
+        if self.asset is not None:
+            asset = self.asset
+        else:
+            asset = self.sensor.generic_asset
+        db_flex_context = asset.flex_context
+        if asset.parent_asset is not None:
+            db_parent_flex_context = asset.parent_asset.flex_context
+        else:
+            db_parent_flex_context = {}
+        self.flex_context = FlexContextSchema().load(
+            {**db_parent_flex_context, **db_flex_context, **self.flex_context}
+        )
```

Comment:
@victorgarcia98 @nhoening rather than a full review, I would like your thoughts on this bit specifically. I involved the flex-context of the parent asset, because 1 test (`test_auth_token_test_inflexible_device_sensors_priority`) was depending on it (for its inflexible device sensors). Do you reckon elevating the search for a relevant flex-context is a good strategy (and maybe extending that further towards the top asset), or should I rather amend the test to not look higher up the tree (for now)?

Reply:
Yes that is relevant. So the idea is that we'd reconstitute the flex-context of the sensor or asset being scheduled, out of any contexts defined upwards in the tree, right? So then I think we'd want to:

- Think about whether this makes sense for each individual flex-context field. (I think it does.)
- Streamline that behavior for all flex-context fields.
- Document that.
- Let users see (UI) the relevant db flex-context that would be reconstituted for any sensor or asset that can be scheduled. Otherwise, it can become difficult to find out the reason behind certain schedules.

Reply:
Resolving this here, since I created a separate issue for the remaining todo.